base/operators.jl

```diff
@@ -351,6 +351,24 @@ for f in (:+, :-)
     end
 end
 
+
+## non-linear operations on ranges and fallbacks for non-real numbers ##
+
+.+(x::Number, r::Range) = [ x+y for y=r ]
```

Comment:
I just took those verbatim from the range file (they need to be moved around for dependency stuff). I do agree that in is more clear for non-range args in general.